libsolidity/interface/FileReader.cpp

```diff
@@ -31,9 +31,23 @@ using std::string;
 namespace solidity::frontend
 {
 
+void FileReader::setBasePath(boost::filesystem::path const& _path)
+{
+	if (_path.empty())
+		m_basePath = "";
+	else
+		m_basePath = normalizeCLIPathForVFS(_path);
+}
+
 void FileReader::setSource(boost::filesystem::path const& _path, SourceCode _source)
 {
-	m_sourceCodes[_path.generic_string()] = std::move(_source);
+	boost::filesystem::path normalizedPath = normalizeCLIPathForVFS(_path).generic_string();
```

Comment:
Why do you call `generic_string` and then re-assign to a `path`?

Reply:
Good catch. That's a mistake. It should not be called here, only below.